src/http11.rs

```diff
@@ -0,0 +1,311 @@
+//! Adapts the HTTP/1.1 implementation into the `HttpMessage` API.
+use std::io::{self, Write, BufWriter, Read};
+use std::net::Shutdown;
+
+use method::{Method};
+use header::{ContentLength, TransferEncoding};
+use header::Encoding::Chunked;
+
+use net::{NetworkConnector, NetworkStream, ContextVerifier};
+use http::{HttpWriter, LINE_ENDING};
+use http::HttpReader::{SizedReader, ChunkedReader, EofReader};
+use http::HttpWriter::{ChunkedWriter, SizedWriter, EmptyWriter};
+use buffer::BufReader;
+use http::{self, HttpReader};
+
+use message::{
+    Protocol,
+    HttpMessage,
+    RequestHead,
+    ResponseHead,
+};
+use header;
+use version;
+
+/// An implementation of the `HttpMessage` trait for HTTP/1.1.
+#[derive(Debug)]
+pub struct Http11Message {
+    stream: Option<Box<NetworkStream + Send>>,
```

Comment:
Fair enough. It could probably be done inside a method on the state enum, but don't worry about it. I might look at if it's possible to keep state similar to the `WriteStatus` that `Request` uses.